### PR TITLE
Big refactoring

### PR DESCRIPTION
**Pull request by macajind**

### CODE REVIEW

File: src/components/ButtonNav/index.tsx [hunk 1]
```diff
@@ -1,43 +1,54 @@
...
-import { Theme, useThemeControl } from '../Theme'
-import { useUser } from '../User'
 
-const NavBeforeLogin = (theme: Theme): React.ReactElement => (
+import { useThemeControl } from 'components/Theme'
+import { useUser } from 'components/User'
+
+import { StyledButton, Wrapper, ButtonLink } from './styled'
+
+const ThemedStyledButton: React.FC<ComponentProps<typeof StyledButton>> = ({
```
Comment (tomas90hampl) [inline src/components/ButtonNav/index.tsx:15]: Nice :slightly_smiling_face:

File: src/components/ButtonNav/styled.ts [hunk 2]
```diff
@@ -18,11 +19,7 @@ export const Wrapper = styled.nav`
 	}
 `
 
-type Props = {
-	color: string
-}
-
-export const StyledButton = styled(Button)<Props>`
+export const StyledButton = styled(Button)<{ $theme: Theme }>`
```
Comment (tomas90hampl) [inline src/components/ButtonNav/styled.ts:22]: What's your opinion for this naming?
```suggestion
export const StyledButton = styled(Button)<{ selectedTheme: Theme }>`
```

File: tsconfig.json [hunk 3]
```diff
@@ -44,7 +44,7 @@
 
     /* Module Resolution Options */
     "moduleResolution": "node",                     /* Specify module resolution strategy: 'node' (Node.js) or 'classic' (TypeScript pre-1.6). */
-    // "baseUrl": "./",                             /* Base directory to resolve non-absolute module names. */
+    "baseUrl": "src",                               /* Base directory to resolve non-absolute module names. */
```
Comment (tomas90hampl) [inline tsconfig.json:47]: Very useful :+1: :slightly_smiling_face:

File: src/components/NavBar/styled.ts [hunk 4]
```diff
@@ -1,7 +1,8 @@
 import styled from 'styled-components'
-import { MediaQueries } from '../../themes'
-import { StyledLink } from '../StyledText/StyledLink'
-import { Theme } from '../Theme'
+import { MediaQueries } from 'themes'
```
Comment (tomas90hampl) [inline src/components/NavBar/styled.ts:2]: one line down :arrow_down: :upside_down_face: Or is it on purpose? MediaQueries is not foreign lib :slightly_smiling_face:

File: src/features/RegisterForm/index.tsx [hunk 5]
```diff
@@ -68,30 +70,32 @@ const RegisterForm: React.FC<Props> = ({ onSuccess }) => {
...
-				await user.updateProfile({ displayName: nickname })
-
-				onSuccess()
-			}
-		} catch (error) {
-			if (error?.code === 'auth/email-already-in-use') {
-				setSubmitError('Tento e-mail již existuje!')
-			} else {
-				console.error(error)
+	const onSubmit = useCallback(
```
Comment (tomas90hampl) [inline src/features/RegisterForm/index.tsx:73]: :eyes: @MartinaVilimova https://reactjs.org/docs/hooks-reference.html#usecallback :eyes:

**Review by tomas90hampl — APPROVED** [review]
Thanks for this refactor :+1:

File: src/components/ButtonNav/index.tsx (hunk 1 above)
Comment (macajind) [reply src/components/ButtonNav/index.tsx:15]: Thank you! :blush:

File: src/components/ButtonNav/styled.ts (hunk 2 above)
Comment (macajind) [reply src/components/ButtonNav/styled.ts:22]: This naming with `$` at the beginning has specific purpose. :sweat_smile:
https://styled-components.com/docs/api#transient-props

File: src/components/NavBar/styled.ts (hunk 4 above)
Comment (macajind) [reply src/components/NavBar/styled.ts:2]: It's just a detail. It looks like a foreign lib, so I left it like this everywhere. Does it really matters so much? :smile:

File: tsconfig.json (hunk 3 above)
Comment (macajind) [reply tsconfig.json:47]: Yes, this is really great for shorting of imports everywhere. :wink:

File: src/components/ButtonNav/styled.ts (hunk 2 above)
Comment (tomas90hampl) [reply src/components/ButtonNav/styled.ts:22]: Ok :)

File: src/components/NavBar/styled.ts (hunk 4 above)
Comment (tomas90hampl) [reply src/components/NavBar/styled.ts:2]: It doesn't matter :)

**Review by MartinaVilimova — APPROVED** [review]

**Comment by MartinaVilimova** [comment]
Great refactor 👍😉

**Pull request merged**